snakemake/script.py

```diff
@@ -484,10 +500,14 @@ def script(
         with tempfile.NamedTemporaryFile(
             suffix="." + os.path.basename(path), dir=dir, delete=False
         ) as f:
-            if not language == "rmarkdown":
-                f.write(preamble.encode())
-                f.write(source)
-            else:
+            if language == "jupyter":
+                nb = nbformat.reads(source, as_version=4)  # nbformat.NO_CONVERT
```

Comment:
I wonder how you envision drafting such notebooks? Since the notebook code has to depend on those variables being defined, it is a bit cumbersome to initially develop it, in particular because one somehow looses the biggest advantage of jupyter, which is its interactiveness. Maybe we can offer the following:
When running Snakemake with

```
snakemake --export-prototype-notebook <rulename>=<notebookfile>
```
It only executes the workflow up to the necessary input of the given rule. Then, it even runs this code if the notebook is not yet present, but it does not execute the notebook. Instead, it just writes a draft file including the preamble. The user can then open that notebook with jupyter, edit it under consideration of the real data coming from the rule, and save it.

Finally, with
```
snakemake --import-prototype-notebook <rulename>=<notebookfile>
```
Snakemake copies the notebook into whatever is defined in the rule, while removing the preamble block from the prototype.

This is not necessary for this PR (it would definitely be a separate PR), but I wanted to hear your thoughts.

Reply:
This point definitely needs some thought. At the moment I develop the notebook outside of Snakemake and use hard-coded paths to existing Snakemake outputs. Once the notebook is "finished", I add it to the pipeline and adjust the paths (using `snakemake.input`, etc). This is certainly far from optimal.

I think to answer this we need to figure out what the central goal of using Notebooks with Snakemake is. For me (so far, I am happy to change my mind 😄) Jupyter notebooks are just another piece of code which computes results (and as a side effect produces a nicely readable PDF with text and figures). Thus my usage of the `script` directive.
If we think of them as a way of building Snakemake pipelines "step-by-step", your proposal might be the way to go.

Reply:
I think it is both. To me, it would kind of optionally replace the normal python or R scripts, allowing for a more interactive way of drafting them. But also in your case, it would simplify the inclusion in the pipeline.

Reply:
Very fascinating use case :)
Maybe a good way of approaching this becomes clearer after the other proposed changes are implemented.